share/picongpu/tests/Pusher/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,364 @@
+
+def compare_radius(x_poss, y_poss, x_offSet, y_offSet, x_momentum, y_momentum,
+                   charge, mass, B, params, num_iter, periode):
+    """Tests if the change in radius from one revolution to the other is
+    greater than epsilon = 1e-5.
+    The radii are calculated by the positions of the particle in the x-y-plane
+    in one test and by the momenta (and q*B) in the x-y-plane in the other one.
+
+    Parameters:
+    x/y_momentum(array): Series of the momenta of the particle in x/y-direction
+    charge (float): Charge of the particle
+    mass( float): Mass of the particle
+    B (float): Magnitude of the magnetic field B in the test
+    params (dict): Simulation parameters
+    num_iter (int): Number of iterations in the series
+    period (int): openPMD period of the simulation
+
+    Returns:
+    compare_result (int):
+        == 0 if the radius of gyration does not vary much (<epsilon)
+        != 0 if the change in radius of gyration large (>epsilon)
+
+    """
+    # the  change in of the radius, measured between 2 turns
+    # is ca. 2e-5 (calculated with position)
+    # Doubling this value yields epsilon as an approximation of the maximal
+    # uncertanty PIConGPU should have when the test uses the position.
+    # Therefore the test uses this value to test the radius change against.
+    # PIConGPU run September 2023
+    epsilon_position = 5e-5
+
+    # calculating gamma for the calculation of the periodic time
+    abs_momentum = np.sqrt(x_momentum[0]**2 + y_momentum[0]**2)
+    gamma = 1/np.sqrt(1 - abs_momentum**2/(abs_momentum**2 + 1))
+    # periodic time / timestep
+    steps_per_revolution = ((2*np.pi * gamma * mass[0]/(abs(charge[0])
+                            * B/params["unit_bfield"])) / params["dt"])
+
+    revolutions_in_series = int(num_iter / steps_per_revolution)
+
+    compare_result_momentum = 0
+    compare_result_positions = 0
+
+    # compare with position
+    radius = np.sqrt((x_poss + x_offSet)**2 + (y_poss + y_offSet)**2)
+
+    # relative comparison
+    for k in range(revolutions_in_series):
+        index_of_revolution = int(steps_per_revolution/periode) * k
+        compare_radii = (
+            abs(radius[index_of_revolution] - radius[index_of_revolution + 1])
+            / radius[index_of_revolution])
+
+        if compare_radii > epsilon_position:
+            print("pusher is not valid (position)")
+            print("please check the simulation")
+            compare_result_positions += 42
+
+        else:
+            print("pusher is valid (position)")
+
+    # the greatest change in of the radius, measured between 2 turns
+    # (2000 Turns simulated) is ca. 4e-6 (calculated with momentum)
+    # Doubling this value yields epsilon as an approximation of the maximal
+    # uncertanty PIConGPU should have when the test uses the momentum.
+    # Therefore the test uses this value to test the radius change against.
+    # PIConGPU run September 2023
+    epsilon_momentum = 1e-5
```

Comment:
As I mentioned above I had to adjust the test because the calculation of the raduis with the position of the particle is not as precice as the calculation with the momentum. Therefore we have to regard two different thresholds for the test to call it a failure. I adjusted the readme.